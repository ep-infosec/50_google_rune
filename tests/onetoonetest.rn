//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class Foo(self) {
  final(self) {
    println "Destroying Foo ", <u32>self
  }
}

class Bar(self, foo: Foo) {
  foo.insertBar(self)
  final(self) {
    println "Destroying Bar ", <u32>self
  }
}

relation OneToOne Foo Bar cascade

foo0 = Foo()
bar0 = Bar(foo0)
foo1 = Foo()
bar1 = Bar(foo1)
bar0.destroy()
foo0 = null(foo0)
foo1 = null(foo1)
