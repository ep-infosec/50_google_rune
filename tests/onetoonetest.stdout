Destroying Bar 0
Destroying Foo 0
Destroying Foo 1
Destroying Bar 1
